steps/create_pdm_trigger.py

```diff
@@ -116,9 +126,68 @@ def get_events_client():
     return boto3.client("events")
 
 
+def get_existing_cloudwatch_event_rules(client):
+    the_logger.info(
+        f"Retrieving all existing PDM cloudwatch rules with prefix of '{CLOUDWATCH_RULE_PREFIX}'",
+    )
+
+    first_batch = client.list_rules(
+        NamePrefix=CLOUDWATCH_RULE_PREFIX,
+    )
+    all_rules = first_batch["Rules"]
+    next_token = (
+        first_batch["NextToken"]
+        if "NextToken" in first_batch
+        else None
+    )
+
+    while next_token is not None:
+        next_batch = client.list_rules(
+            NamePrefix=CLOUDWATCH_RULE_PREFIX,
+            NextToken=next_token,
+        )
+        all_rules.extend(next_batch["Rules"])
+        next_token = (
+            next_batch["NextToken"]
+            if "NextToken" in next_batch
+            else None
+        )
+
+    the_logger.info(
+        f"Retrieved {len(all_rules)} existing PDM cloudwatch rules with prefix of '{CLOUDWATCH_RULE_PREFIX}'",
+    )
+
+    return list({ rule['Name'] : rule for rule in all_rules }.values())
```

Comment:
Not really sure what this is achieving:
![image](https://user-images.githubusercontent.com/52243130/117177017-743ab100-adc8-11eb-8eb1-91b00c88d898.png)

Reply:
It de-duplicates the list. You can see in one of the unit tests it ensures that the list is de-duplicated. With python as it's a list of json dicts, I can't just turn it in to a set as it doesn't know what constitutes equality.